engine/src/main/java/org/terasology/logic/characters/CharacterSystem.java

```diff
@@ -82,9 +83,13 @@
     @In
     private BlockEntityRegistry blockRegistry;
 
-    @ReceiveEvent(components = {CharacterComponent.class})
-    public void onDeath(DoDestroyEvent event, EntityRef entity) {
-        CharacterComponent character = entity.getComponent(CharacterComponent.class);
+    @ReceiveEvent
+    public void beforeDestroy(BeforeDestroyEvent event, EntityRef entity, CharacterComponent character) {
+        if (character.controller != EntityRef.NULL) {
```

Comment:
This looks wrong. A destroyed entity will also behave like EntityRef.NULL but will have different id.